### DISCUSSION TITLE
CMake Support

### DISCUSSION THREAD

**Discussion by 1ynxy** [CuarzoSoftware/Louvre#56] [Support] [answered]
I'm going to out myself as a dumb little guy here, but I'm struggling to import this as a system library using CMake on an Arch system- the linking for requirements appears to be all mussed up. Is this possible? Can you provide any instructions or tips? I'm afraid I've very little experience with Meson.

**Comment by ehopperdietzel** [comment, marked as answer]
Have you installed it yet? If not, you can easily get it via [AUR](https://aur.archlinux.org/packages/louvre). Additionally, if you're interested in using the latest version, say the 2.0.0 branch, you'll need to build and install it manually using Meson.

I'm not very familiar with CMake, but I believe this is how you can link it to your project:

```cmake
project(your_project)

...

find_package(PkgConfig REQUIRED)
pkg_check_modules(LOUVRE REQUIRED Louvre)
include_directories(${LOUVRE_INCLUDE_DIRS})
link_directories(${LOUVRE_LIBRARY_DIRS})
add_definitions(${LOUVRE_CFLAGS_OTHER})
target_link_libraries(your_project ${LOUVRE_LIBRARIES})
```

The lib name is `libLouvre.so` and the headers directory must be included e.g `/usr/local/include/Louvre`.

However, I'd recommend using Meson instead. Here's an example of a Louvre project using Meson: [LouvreTutorial](https://github.com/CuarzoSoftware/LouvreTutorial/tree/main). The `meson.build` file is quite straightforward.

**Reply by 1ynxy** [reply]
You're brilliant- that does indeed build just fine. And yup, I had at least managed to install Louvre using AUR ;)

I'll try Meson at some point- I'm just building in some other components that I'm used to building using CMake haha. Thanks!
